src/plugins/intel_gpu/src/kernel_selector/kernels/permute/permute_kernel_f_y_axes.cpp

```diff
@@ -0,0 +1,234 @@
+    const size_t feature_block_size = GetFeatureBlockSize(params);
+    jit.AddConstant(MakeJitConstant("BLOCK_SIZE", tile_width));
+    jit.AddConstant(MakeJitConstant("VEC_SIZE", vector_size));
+    jit.AddConstant(MakeJitConstant("J_TIMES", j_times));
+    jit.AddConstant(MakeJitConstant("TILE_SIZE", tile_size));
+    jit.AddConstant(MakeJitConstant("FEATURE_BLOCK_SIZE", feature_block_size));
+
+    const auto layout = params.inputs.front().GetLayout();
+    const auto subgroup_size = Is3DTranspose(params) ? feature_block_size : tile_size;
+    if (IsSIMDSizeSupported(params.engineInfo, subgroup_size) &&
+        (layout == DataLayout::b_fs_yx_fsv32 || layout == DataLayout::b_fs_yx_fsv16)) {
+        jit.AddConstant(MakeJitConstant("SUB_GROUP_SIZE", subgroup_size));
+    }
+
+    if (!params.fused_ops.empty()) {
+        const std::vector<std::string> original_output_order = {"b_idx, f_idx, y_idx, x_idx"};
+        const FusedOpsConfiguration conf = {"", original_output_order, "res", params.inputs[0].GetDType(), 1};
+        jit.Merge(MakeFusedOpsJitConstants(params, {conf}));
+    }
+    return jit;
+}
+
+static inline std::vector<size_t> GetGWS(const permute_params& params) {
+    const auto& in = params.inputs[0];
+    std::vector<size_t> gws;
+    auto block_size = IsSimpleMemCopyOperation(params) ? GetTileWidth(params) : GetTileSize(params);
+    if (params.inputs[0].X().v == 1) {
+        gws = {in.X().v, in.Y().v / block_size, (in.Batch().v * in.Feature().v)};
+    } else {
+        if (Is3DTranspose(params)) {
+            gws = {in.X().v / block_size, in.Y().v / GetFeatureBlockSize(params), (in.Batch().v * in.Feature().v)};
+        } else {
+            gws = {in.X().v / block_size, in.Y().v, (in.Batch().v * in.Feature().v)};
+        }
+    }
+    return gws;
+}
+
+CommonDispatchData PermuteKernel_f_y_axes::SetDefault(const permute_params& params) const {
+    CommonDispatchData dispatchData;
+    dispatchData.gws = GetGWS(params);
+    if (IsSimpleMemCopyOperation(params)) {
+        auto in_layout = params.inputs[0].GetLayout();
+        auto out_layout = params.outputs[0].GetLayout();
+        const std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws = {
+            {Tensor::DataChannelName::X},
+            {Tensor::DataChannelName::Y},
+            {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
+        dispatchData.lws =
+            GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo, in_layout, out_layout, dims_by_gws);
+    } else if (Is3DTranspose(params)) {
+        dispatchData.lws = {1, 1, GetFeatureBlockSize(params)};
+    } else {
+        dispatchData.lws = {1, 1, GetTileSize(params)};
+    }
+    return dispatchData;
+}
+
+bool PermuteKernel_f_y_axes::Validate(const Params& p, const optional_params& o) const {
+    if (!Parent::Validate(p, o)) {
+        return false;
+    }
+
+    const auto is_rotating_except_batch = [](const std::vector<uint16_t>& order) {
```

Comment:
Please change this  function name, originally the "rotating" intended such transpose like [bfyx=>byxf, bfzyx => bzyxf]
where the dimensions are "rotating" except "batch" dim. 
However the target pattern of this kernel does not "rotating", it is rather, swapping ?

Reply:
Changed for better naming.